### PR TITLE
Remove CPU and memory query constants

### PR DESCRIPTION
**Pull request by shreyabiradar07**
## Description

This PR has the following changes:

- Removes CPU and memory queries from `DataSourceQueries.java` class to MetricProfile YAML
- Throws `NullPointerException` when input `maxDate` query is missing or null in `generateRecommendations` codeflow

### Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Docs update
- [ ] Breaking change (What changes might users need to make in their application due to this PR?)
- [ ] Requires DB changes

## How has this been tested?

Tested using local monitoring demo script

**Test Configuration**
* Kubernetes clusters tested on: Openshift (ResourceHub cluster)

## Checklist :dart:

- [x] Followed coding guidelines
- [x] Comments added
- [ ] Dependent changes merged
- [ ] Documentation updated
- [ ] Tests added or updated

## Additional information

Image:  quay.io/shbirada/mvp_demo:remove-query-constants-v1

### CODE REVIEW

File: src/main/java/com/autotune/analyzer/performanceProfiles/MetricProfileCollection.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        LOGGER.info(KruizeConstants.MetricProfileConstants.ADDING_METRIC_PROFILE + "{}", metricProfileName);
+
+        if(metricProfileCollection.containsKey(metricProfileName)) {
+            LOGGER.error(KruizeConstants.MetricProfileConstants.METRIC_PROFILE_ALREADY_EXISTS + "{}", metricProfileName);
+        } else {
+            LOGGER.info(KruizeConstants.MetricProfileConstants.METRIC_PROFILE_ADDED + "{}", metricProfileName);
+            metricProfileCollection.put(metricProfileName, metricProfile);
+        }
+    }
+}
```
Comment (msvinaykumar) [inline src/main/java/com/autotune/analyzer/performanceProfiles/MetricProfileCollection.java:75]: EOF missing

File: src/main/java/com/autotune/database/table/KruizeMetricProfileEntry.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    public JsonNode getSlo() {
+        return slo;
+    }
+
+    public void setSlo(JsonNode slo) {
+        this.slo = slo;
+    }
+}
```
Comment (msvinaykumar) [inline src/main/java/com/autotune/database/table/KruizeMetricProfileEntry.java:107]: EOF missing

**Comment by msvinaykumar** [comment]
Review Checklist

#### Common Review
- [ ] Code formatting adheres to the project’s style guide.
- [ ] No unnecessary commented-out code.
- [ ] Appropriate use of comments and documentation.

#### Logical Flow Review
- [ ] Code logic is clear and easy to follow.
- [ ] All branches and conditions are covered and tested.
- [ ] No redundant or duplicate code.

#### Demo Update
- [ ] Demonstration examples are updated to reflect changes.
- [ ] README and documentation include updated demo instructions.
- [ ] Any necessary demo files are included and tested.

#### Naming Conventions
- [ ] Variable and function names follow the project's naming conventions.
- [ ] File names are descriptive and adhere to naming standards.
- [ ] No ambiguous or misleading names.

#### Workflow
- [ ] CI/CD pipelines are updated and passing.
- [ ] Tests are included and all pass.
- [ ] The workflow for deploying or integrating changes is clear and documented.

#### Does It Affect ROS 
- [ ] Changes are reviewed for compatibility with ROS.
- [ ] ROS-specific configurations are updated if necessary.
- [ ] Any ROS dependencies or integrations are tested and verified.

**Review by msvinaykumar — APPROVED** [review]
LGTM

**Review by chandrams — APPROVED** [review]
LGTM

**Pull request merged**